python/pyspark/pandas/tests/test_series.py

```diff
@@ -394,6 +394,21 @@ def test_isin(self):
         with self.assertRaisesRegex(TypeError, msg):
             psser.isin(1)
 
+        # when Series have NaN
+        pser = pd.Series(["lama", "cow", None, "lama", "beetle", "lama", "hippo", None], name="a")
+        psser = ps.from_pandas(pser)
+
+        self.assert_eq(psser.isin(["cow", "lama"]), pser.isin(["cow", "lama"]))
+
+        pser = pd.Series([None, 5, None, 3, 2, 1, None, 0, 0], name="a")
+        psser = ps.from_pandas(pser)
+
+        if LooseVersion(pd.__version__) >= LooseVersion("1.2"):
+            self.assert_eq(psser.isin([1, 5, 0, None]), pser.isin([1, 5, 0, None]))
+        else:
+            expected = pd.Series([True, True, True, False, False, True, True, True, True], name="a")
```

Comment:
This still fails.

```
Series values are different (33.33333 %)
[index]: [0, 1, 2, 3, 4, 5, 6, 7, 8]
[left]:  [False, True, False, False, False, True, False, True, True]
[right]: [True, True, True, False, False, True, True, True, True]
```

The `expected` should:
```suggestion
            expected = pd.Series([False, True, False, False, False, True, False, True, True], name="a")
```
?

Reply:
Thank you, fixed. Please take another look when you have time.